ERR_PNPM_LINKING_FAILED when running pnpm i

### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

8.12.1 - 8.13.1

### Which area(s) of pnpm are affected? (leave empty if unsure)

_No response_

### Link to the code that reproduces this issue or a replay of the bug

_No response_

### Reproduction steps

Updating for instance nx in my repo, then running pnpm i to install the new deps often fails with 

ERR_PNPM_LINKING_FAILED  Error: The filename, directory name, or volume label syntax is incorrect. (os error -2147024773), reflink...

running pnpm i again, after the failure usually works...

### Describe the Bug

When updating package.json and then running pnpm i on windows, the last two versions often fail with 
```
 ERR_PNPM_LINKING_FAILED  Error: The filename, directory name, or volume label syntax is incorrect. (os error -2147024773), reflink 'D:\.pnpm-store\v3\files\aa\5bab73d5f67e005dbe01f7f5aec6b9e7b03dfa3d300507b2056c2c82de9fae75ebce3cdbc3c28bf2e8311e3219b05802d115d854f88689014c78b09454c00b' -> 'D:\repos\teve-no\task4s\node_modules\.pnpm\@nx+eslint-plugin@17.2.8_@swc-node+register@1.6.8_@swc+core@1.3.101_@types+node@20.10.5_@type_alsrajq2zdvzzqi5j3o5xsqlle\node_modules\@nx\eslint-plugin_tmp_16432\src\migrations\update-17-2-6-rename-workspace-rules\rename-workspace-rules.js'
```

The error is resolved if running pnpm i again...

The error is not present in 8.12.0 it seems, but the error is not completely deterministic, so it might be present there as well (but have so far not encountered it with 8.12.0)...

### Expected Behavior

No error when running pnpm i

### Which Node.js version are you using?

20.10.0

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

_No response_

--- a/.changeset/quiet-ladybugs-whisper.md
+++ b/.changeset/quiet-ladybugs-whisper.md
@@ -1,0 +1,6 @@
+---
+"@pnpm/fs.indexed-pkg-importer": patch
+"pnpm": patch
+---
+
+Fix copy-on-write on Windows Dev Drives [#7468](https://github.com/pnpm/pnpm/issues/7468).

--- a/fs/indexed-pkg-importer/package.json
+++ b/fs/indexed-pkg-importer/package.json
@@ -18,7 +18,7 @@
     "@pnpm/core-loggers": "workspace:*",
     "@pnpm/graceful-fs": "workspace:*",
     "@pnpm/store-controller-types": "workspace:*",
-    "@reflink/reflink": "0.1.15",
+    "@reflink/reflink": "0.1.16",
     "@zkochan/rimraf": "^2.1.3",
     "fs-extra": "^11.1.1",
     "make-empty-dir": "^2.1.0",

